src/common/src/util/value_encoding/mod.rs

```diff
@@ -44,6 +44,28 @@ pub fn deserialize_cell(mut data: impl Buf, ty: &DataType) -> Result<Datum> {
     deserialize_value(ty, &mut data)
 }
 
+/// Serialize a datum into bytes (Not order guarantee, used in value encoding).
+pub fn serialize_datum(cell: &Datum) -> Result<Vec<u8>> {
```

Comment:
You may rename it to `serialize_cell` and change the origin one into `serialize_cell_not_null` which we used to have before my commit and we think the null tag is useless and remove it🤣.

Reply:
I think we should keep the current names. Current `serialize_cell` actually can serialize a datum into an empty vec, which is the expected behavior in cell based context. And the  `serialize_datum` function will never be used in cell based context, it only loyally encodes the datum.

Reply:
Not sure what's the diff between `serialize_cell` and `serialize_datum` in your understanding? I think this two function is hard to tell the difference from name. May also add some function comments?